pep-0XXX-environment-markers.rst

```diff
@@ -0,0 +1,171 @@
+between them may mean that particular dependencies are only needed on
+particular platforms (relying on ``pywin32`` only on Windows, for
+example)"
+
+Environment Markers attempt to provide more flexibility in a list of
+requirements by allowing the developer to list requirements that are
+specific to a particular environment.
+
+Examples
+========
+
+Here are some examples of such markers inside a requirements.txt::
+
+   pywin32 >=1.0 ; sys_platform == 'win32'
+   unittest2 >=2.0,<3.0 ; python_version == '2.4' or python_version == '2.5'
+   backports.ssl_match_hostname >= 3.4 ; python_version < '2.7.9' or (python_version >= '3.0' and python_version < '3.4')
+   hawkey ; 'linux' in sys_platform
+
+And here's an example of some conditional metadata included in
+setup.py for a distribution that requires PyWin32 both at runtime and
+buildtime when using Windows::
+
+   setup(
+     install_requires=["pywin32 > 1.0 : sys.platform == 'win32'"],
+     setup_requires=["pywin32 > 1.0 : sys.platform == 'win32'"]
+     )
+
+
+Micro-language
+==============
+
+The micro-language behind this is as follows. It compares::
+
+* strings with the ``==`` and ``in`` operators (and their opposites)
+* version numbers with the ``<``, ``<=``, ``>=``, and ``<`` operators
+  in addition to those supported for strings
+
+The usual boolean operators ``and`` and ``or`` can be used to combine
+expressions, and parentheses are supported for grouping.
+
+The pseudo-grammar is ::
+
+    MARKER: EXPR [(and|or) EXPR]*
+    EXPR: ("(" MARKER ")") | (STREXPR|VEREXPR)
+    STREXPR: STRING [STRCMPOP STREXPR]
+    STRCMPOP: ==|!=|in|not in
+    VEREXPR: VERSION [VERCMPOP VEREXPR]
+    VERCMPOP: (==|!=|<|>|<=|>=)
+
+
+``SUBEXPR`` is either a Python string (such as ``'win32'``) or one of
+the ``Strings`` marker variables listed below.
+
+``VEREXPR`` is a PEP-0440[3] version identifier, or one of the
+``Version number`` marker variables listed below. Comparisons between
+version numbers are done using PEP-0440 semantics.
+
+
+Strings
+-------
+
+* ``os_name``: ``os.name``
+* ``sys_platform``: ``sys.platform``
+* ``platform_release``: ``platform.release()``
+* ``implementation_name``: ``sys.implementation.name``
+* ``platform_machine``: ``platform.machine()``
+* ``platform_python_implementation``: ``platform.python_implementation()``
+
+
+If a particular string value is not available (such as ``sys.implementation.name``
+in versions of Python prior to 3.3), the corresponding marker
+variable MUST be considered equivalent to the empty string.
+
+If a particular version number value is not available (such as
+``sys.implementation.version`` in versions of Python prior to 3.3) the
+corresponding marker variable MUST be considered equivalent to ``0``
+
+
+Version numbers
+---------------
+
+* ``python_version``: ``platform.python_version()[:3]``
+* ``python_full_version``: see definition below
+* ``platform_version``: ``platform.version()``
```

Comment:
This actually returns a string:

>>> print platform.version()
Darwin Kernel Version 14.5.0: Wed Jul 29 02:18:53 PDT 2015; root:xnu-2782.40.9~2/RELEASE_X86_64